### PR TITLE
2439 use hover as fallback for main menu

### LINKED ISSUE #2439: Use hover as fallback on desktop platforms
While we _need_ `onClick` on mobile devices for expanding and collapsing the main menu on mobile devices and tablets, we can have `hover` as a fallback on desktop when the JavaScript code fails.

Here is how I foresee the implementation.

- Add a block to the main menu CSS that is scoped to the class `nojs`.
- Inside this block add the relevant styling that will allow the menu to open and close on hover
- Update the React JavaScript code to remove the `nojs` class when attaching its event handler

This way a user can still expand and interact with the main menu if there is an error that causes the JavaScript to fail but, if all is well, the menu interaction will be as it is currently implemented.

Part of https://github.com/mdn/yari/issues/1124

### PR DESCRIPTION
**Pull request by schalkneethling**
- fix: use `:hover` and `:focus` as fallback on small desktop viewports and up
- ensure focus state also opens menu without JS

NOTE: You need to test this via port `:5000` as disabling JavaScript on port `3000` results in a blank page. I assume when on port `3000` it bypasses server-side rendering.

fix #2439

### CODE REVIEW

File: client/src/ui/molecules/main-menu/index.scss [hunk 1]
```diff
@@ -141,6 +143,18 @@ ul.main-menu {
   }
 }
 
+/* enable hover interaction if javascript not available */
```
Comment (schalkneethling) [inline client/src/ui/molecules/main-menu/index.scss:146]:
```suggestion
/* enable hover interaction if javascript is not available */
```

File: client/src/ui/molecules/main-menu/index.tsx [hunk 2]
```diff
@@ -66,6 +67,18 @@ export default function MainMenu({
...
   useEffect(() => {
+    const mainMenu = mainMenuRef.current;
+
+    // by default the main menu contains a `nojs` class which
+    // then allows users on desktop to interact with the main
+    // menu via hover events if the JavsScript failed for whatever
+    // reason. If all is well though, we remove the class here and
+    // let JavaScript take over the interaction
+    if (mainMenu) {
+      console.log("remove nojs");
```
Comment (schalkneethling) [inline client/src/ui/molecules/main-menu/index.tsx:78]:
```suggestion

```

File: client/src/ui/molecules/main-menu/index.scss [hunk 3]
```diff
@@ -73,17 +73,19 @@ ul.main-menu {
 
   .top-level-entry-container {
     position: relative;
-  }
 
-  ul.show {
-    display: block;
-
-    @media #{$mq-tablet-and-up} {
+    ul {
```
Comment (fiji-flo) [inline client/src/ui/molecules/main-menu/index.scss:77]: is this intended, this is quite a design change for mobile?

File: client/src/ui/molecules/main-menu/index.scss [hunk 4]
```diff
@@ -141,6 +143,18 @@ ul.main-menu {
   }
 }
 
+/* enable hover interaction if javascript is not available */
+@media #{$mq-tablet-and-up} {
```
Comment (fiji-flo) [inline client/src/ui/molecules/main-menu/index.scss:147]: Looks like it's working.

**Review by fiji-flo — APPROVED** [review]
👍

**Pull request merged**